### DISCUSSION TITLE
onClick in Components

### DISCUSSION THREAD

**Discussion by DIGnmg** [vercel/next.js#12955] [Help] [answered]
Hello! This should be pretty easy but i can not seem to find an answer or get it to work. I have a component that is inside of the page. Inside of this component i have an onClick event. I can not get this event to trigger. Wont call a hook, or console log or anything. I have turned off SSR to see if it was just bc of the events were not working bc it was being rendered on the server or what. Any tips would help.

Thanks

**Comment by rafaelalmeidatk** [comment, marked as answer]
Hello! Can you share the code of this component? Next shouldn't be affecting these events

**Reply by DIGnmg** [reply]
I am using firebase for Auth. And looks like when I call 'firebase.auth().currentUser' in the component it freezes it and I can not use the event handlers. But if I do not call that in the app. It seems to work. Its odd.

**Reply by DIGnmg** [reply]
looks like bad config.... I might update my help request. Is there a better way to handle env vars for nextjs client app?

**Reply by rafaelalmeidatk** [reply]
I don't know what you are doing now but check this doc page: https://nextjs.org/docs/basic-features/environment-variables

**Comment by jensmeindertsma** [comment]
Hey @DIGnmg ! I can help you solve this if you provide some code of where the problem is coming from. If you don't know where it is, perhaps you could create a repository and share that. I'm looking forward to helping you further as soon as possible

**Reply by DIGnmg** [reply]
I got it working. The issue was with my env files and the firebase module. Gonna mark this one as done!

Thanks everyone!
